apertium/latex_accentsmap.cc

```diff
@@ -118,95 +118,104 @@ void AccentsMap::init_locale(){
-	map[L"_"] = L"\\_";
-	map[L"≥"] = L"$\\ge$";   # Math operators
-	map[L"≤"] = L"$\\le$";
-	map[L"≠"] = L"$\\neq$";
-	map[L"©"] = L"\copyright"; # Misc
-	map[L"ı"] = L"{\\i}";
-	map[L"µ"] = L"$\\mu$";
-	map[L"°"] = L"$\\deg$";
-	map[L"‘"] = L"`";    #Quotes
-	map[L"’"] = L"'";
-	map[L"“"] = L"``";
-	map[L"”"] = L"''";
-	map[L"‚"] = L",";
-	map[L"„"] = L",,";
+	map["à"_u] = "\\`a"_u; # Grave accent
+	map["è"_u] = "\\`e"_u;
+	map["ì"_u] = "\\`\\i"_u;
+	map["ò"_u] = "\\`o"_u;
+	map["ù"_u] = "\\`u"_u;
+	map["ỳ"_u] = "\\`y"_u;
+	map["À"_u] = "\\`A"_u;
+	map["È"_u] = "\\`E"_u;
+	map["Ì"_u] = "\\`\\I"_u;
+	map["Ò"_u] = "\\`O"_u;
+	map["Ù"_u] = "\\`U"_u;
+	map["Ỳ"_u] = "\\`Y"_u;
+	map["á"_u] = "\\'a"_u; # Acute accent
+	map["é"_u] = "\\'e"_u;
+	map["í"_u] = "\\'\\i"_u;
+	map["ó"_u] = "\\'o"_u;
+	map["ú"_u] = "\\'u"_u;
+	map["ý"_u] = "\\'y"_u;
+	map["Á"_u] = "\\'A"_u;
+	map["É"_u] = "\\'E"_u;
+	map["Í"_u] = "\\'\\I"_u;
+	map["Ó"_u] = "\\'O"_u;
+	map["Ú"_u] = "\\'U"_u;
+	map["Ý"_u] = "\\'Y"_u;
+	map["â"_u] = "\\^a"_u; # Circumflex
+	map["ê"_u] = "\\^e"_u;
+	map["î"_u] = "\\^\\i"_u;
+	map["ô"_u] = "\\^o"_u;
+	map["û"_u] = "\\^u"_u;
+	map["ŷ"_u] = "\\^y"_u;
+	map["Â"_u] = "\\^A"_u;
+	map["Ê"_u] = "\\^E"_u;
+	map["Î"_u] = "\\^\\I"_u;
+	map["Ô"_u] = "\\^O"_u;
+	map["Û"_u] = "\\^U"_u;
+	map["Ŷ"_u] = "\\^Y"_u;
+	map["ä"_u] = "\\\"a"_u;    # Umlaut or dieresis
+	map["ë"_u] = "\\\"e"_u;
+	map["ï"_u] = "\\\"\\i"_u;
+	map["ö"_u] = "\\\"o"_u;
+	map["ü"_u] = "\\\"u"_u;
+	map["ÿ"_u] = "\\\"y"_u;
+	map["Ä"_u] = "\\\"A"_u;
+	map["Ë"_u] = "\\\"E"_u;
+	map["Ï"_u] = "\\\"\\I"_u;
+	map["Ö"_u] = "\\\"O"_u;
+	map["Ü"_u] = "\\\"U"_u;
+	map["Ÿ"_u] = "\\\"Y"_u;
+	map["ç"_u] = "\\c{c}"_u;   # Cedilla
+	map["Ç"_u] = "\\c{C}"_u;
+	map["œ"_u] = "{\\oe}"_u;   # Ligatures
+	map["Œ"_u] = "{\\OE}"_u;
+	map["æ"_u] = "{\\ae}"_u;
+	map["Æ"_u] = "{\\AE}"_u;
+	map["å"_u] = "{\\aa}"_u;
+	map["Å"_u] = "{\\AA}"_u;
+	map["–"_u] = "--"_u;   # Dashes
+	map["—"_u] = "---"_u;
+	map["ø"_u] = "{\\o}"_u;    # Misc latin-1 letters
+	map["Ø"_u] = "{\\O}"_u;
+	map["ß"_u] = "{\\ss}"_u;
+	map["¡"_u] = "{!`}"_u;
+	map["¿"_u] = "{?`}"_u;
+	map[L"\\"] = "\\\\"_u;    # Characters that should be quoted
+	map["~"_u] = "\\~"_u;
+	map["&"_u] = "\\&"_u;
+	map["$"_u] = "\\$"_u;
+	map["{"_u] = "\\{"_u;
+	map["}"_u] = "\\}"_u;
+	map["%"_u] = "\\%"_u;
+	map["#"_u] = "\\#"_u;
+	map["_"_u] = "\\_"_u;
+	map["≥"_u] = "$\\ge$"_u;   # Math operators
+	map["≤"_u] = "$\\le$"_u;
+	map["≠"_u] = "$\\neq$"_u;
+	map["©"_u] = "\copyright"_u; # Misc
+	map["ı"_u] = "{\\i}"_u;
+	map["µ"_u] = "$\\mu$"_u;
+	map["°"_u] = "$\\deg$"_u;
+	map["‘"_u] = "`"_u;    #Quotes
+	map["’"_u] = "'"_u;
+	map["“"_u] = "``"_u;
+	map["”"_u] = "''"_u;
+	map["‚"_u] = ","_u;
+	map["„"_u] = ",,"_u;
 ]*/
 
 
+void fputus(const UString& s, FILE* out)
+{
+  string temp;
+  temp.reserve(s.size()*2);
+  utf8::utf16to8(s.begin(), s.end(), std::back_inserter(temp));
+  fputs(temp.c_str(), out);
+}
 
-
-
-
+void fputrange(const char* s, int start, int len, FILE* out)
```

Comment:
Surely this function is just `fwrite()` ?

Reply:
it was outputting substrings and I didn't think the indexing numbers were guaranteed to be constant after converting to `ustring`.